lib/module.js

```diff
@@ -334,7 +347,8 @@ Module._resolveLookupPaths = function(request, parent, newReturn) {
   }
 
   // Check for relative path
-  if (request.charCodeAt(0) !== 46/*.*/ &&
+  if (request.length < 2 ||
+      request.charCodeAt(0) !== 46/*.*/ ||
       (request.charCodeAt(1) !== 46/*.*/ &&
        request.charCodeAt(1) !== 47/*/*/)) {
     var paths = modulePaths;
```

Comment:
The use of `charCodeAt` jazz really complicates the logic. It was introduced in a pass to optimize lots of things but I suspect it was micro-opts carried a bit too far. Anyways, would this do to fix it:

```js
! (request.charCodeAt(0) === codeOfDot &&
    (request.charCodeAt(1) === codeOfDot || request.charCodeAt(1) === codeOfSlash))
```